File: kotlin/src/main/kotlin/com/joshy/hywrapper/model/skyblock/ProfilesResponse.kt
package com.joshy.hywrapper.model.skyblock

import com.joshy.hywrapper.model.HypixelResponse
import com.joshy.hywrapper.model.RateLimit
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

/**
 * Information about a player's profiles
 *
 * @property success Whether the request was successful.
 * @property cause The cause of the error, if the request failed.
 * @property profiles A list of the player's profiles
 * */
@Serializable
data class ProfilesResponse(
    override val success: Boolean? = null,
    override val cause: String? = null,
    val profiles: List<Profile>? = null,
) : HypixelResponse {
    @Transient
    override var rateLimit: RateLimit? = null
}

File: python/src/hywrapper/models/skyblock/profiles_response.py
from __future__ import annotations

from typing import List, Optional

from pydantic import ConfigDict, Field

from hywrapper.models.hypixel_response import HypixelResponse
from hywrapper.models.rate_limit import RateLimit
from hywrapper.models.skyblock.profile_response import Profile


class ProfilesResponse(HypixelResponse):
    """
    Information about a player's profiles

    Attributes:
        success: Whether the request was successful.
        cause: The cause of the error, if the request failed.
        profiles: A list of the player's profiles
    """

    model_config = ConfigDict(populate_by_name=True)
    profiles: Optional[List[Profile]] = Field(default=None)
    rate_limit: Optional[RateLimit] = Field(default=None, exclude=True, alias="rateLimit")
